net/acme/files/run.sh

```diff
@@ -211,7 +222,12 @@ issue_cert()
     set -- $domains
     main_domain=$1
 
-    [ -n "$webroot" ] || [ -n "$dns" ] || pre_checks "$main_domain" || return 1
+    if [ -n "$user_setup" ] && [ -f "$user_setup" ]; then
+        log "Running user-provided setup script from $user_setup."
+        "$user_setup" "$main_domain" || return 1
+    else
+        [ -n "$webroot" ] || [ -n "$dns" ] || pre_checks "$main_domain" || return 1
```

Comment:
Oh, I meant the "don't override the default behaviour" comment to also apply to the pre checks :)

Reply:
Hmm... from a quick glance this looks to be easier said than done. The built-in pre_checks does `return 1` if it detects something listening on port 80 that is not uhttpd or nginx.

Idea behind the user provided scripts is to provide freedom to setup whatever the user wishes to serve the challenges (or to route the request elsewhere etc. etc.) and that becomes difficult if we have to try to avoid the detection logic in pre_checks without overriding it.

Would you be willing to re-consider going with the override behavior? Or do you have another idea how to approach this?